### PR TITLE
Added a new flag for pending collectives

### PR DESCRIPTION
**Pull request by yellowwoods12**
In reference to this issue : opencollective/opencollective#1878

### CODE REVIEW

**Comment by yellowwoods12** [comment]
I have added a new flag but  I am not sure where do I have to add the function to update it accordingly?

**Comment by znarf** [comment]
- approvedAt will be set the the current date when the collective
  - is approved by the host: https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/collectives.js#L440
  - automatically approved (GitHub flow): https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/collectives.js#L176

- approvedAt will be set to NULL when
  - the collective change the host: https://github.com/opencollective/opencollective-api/blob/master/server/models/Collective.js#L1502
  - or is archived: https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/collectives.js#L614

**Comment by yellowwoods12** [comment]
>     * approvedAt will be set the the current date when the collective
>       
>       * is approved by the host: https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/collectives.js#L440
>       * automatically approved (GitHub flow): https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/collectives.js#L176
> 
>     * approvedAt will be set to NULL when
>       
>       * the collective change the host: https://github.com/opencollective/opencollective-api/blob/master/server/models/Collective.js#L1502
>       * or is archived: https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/collectives.js#L614

Hi @znarf, I've updated the flag values according to this. Can you have a look at it?

File: server/models/Collective.js [hunk 1]
```diff
@@ -308,6 +308,11 @@ export default function(Sequelize, DataTypes) {
         defaultValue: false,
       },
 
+      approvedAt: {
+        type: DataTypes.DATE,
+        defaultValue: Sequelize.NOW,
+      },
+
```
Comment (znarf) [inline server/models/Collective.js]: In the spec, we said we wanted a behavior similar `isArchived`.

We wanted to expose `isApproved` (a boolean), not `approvedAt` (a date).

**Comment by yellowwoods12** [comment]
@znarf, I've added both isApproved(boolean) and approvedAt(Date) because it was mentioned in the issue that approvedAt will be set with date values. Is it fine now?

**Comment by znarf** [comment]
No it's not, you should not mistake the database model and what is exposed in GraphQL. Looks at how `isArchived`/`deactivatedAt` is working. We want the same with 'isApproved'/`approvedAt`.

- in the database: `approvedAt`
- in GraphQL: `isApproved`

**Comment by yellowwoods12** [comment]
@znarf  @Betree Please review.

File: server/models/Collective.js (hunk 1 above)
Comment (Betree) [reply server/models/Collective.js]: @znarf But this is the model, not the graphql endpoint.

Comment (Betree) [reply server/models/Collective.js]: @znarf But this is the model, not the graphql endpoint.

**Comment by yellowwoods12** [comment]
@Betree Is it fine now?

File: server/graphql/v1/mutations/collectives.js [hunk 2]
```diff
@@ -675,7 +675,7 @@ export async function archiveCollective(_, args, req) {
     throw new Error('Cannot archive collective with balance > 0');
   }
 
-  return collective.update({ isActive: false, deactivatedAt: Date.now() });
+  return collective.update({ isActive: false, deactivatedAt: Date.now(), approvedAt: NULL });
```
Comment (Betree) [inline server/graphql/v1/mutations/collectives.js]: Does this work?

File: server/models/Collective.js [hunk 3]
```diff
@@ -1510,6 +1516,7 @@ export default function(Sequelize, DataTypes) {
     }
     this.HostCollectiveId = null;
     this.isActive = false; // we should rename isActive to isApproved (by the host)
+    this.approvedAt = NULL;
```
Comment (Betree) [inline server/models/Collective.js]: Does this work?

**Comment by Betree** [comment]
@yellowwoods12 Please be aware that since https://github.com/opencollective/opencollective/issues/1878#issuecomment-508779197 this bounty is no more locked for you so someone else could take it.

**Comment by yellowwoods12** [comment]
> @yellowwoods12 Please be aware that since [opencollective/opencollective#1878 (comment)](https://github.com/opencollective/opencollective/issues/1878#issuecomment-508779197) this bounty is no more locked for you so someone else could take it.

@Betree okay. Should I stop working on this issue then?

**Comment by Betree** [comment]
@yellowwoods12 If it's important to you to complete it we could lock it one more week but I recommend that you take a smaller issue until you feel confident with Open Collective's stack, something that only require changes to the frontend. We can only provide limited support and there's still some work to do on this one.

**Comment by znarf** [comment]
Thank you for the effort. We unfortunately had to move forward with another implementation. https://github.com/opencollective/opencollective-api/pull/2370

**Pull request closed without merging**